### ISSUE TITLE
Exception in API - Certificates

### ISSUE THREAD

**Issue by willseward** [rancher/rancher#8610]
**Rancher Versions:**
Server: 1.5.5

**Docker Version:** 1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):** Ubuntu, Vultr

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)** External

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** Cattle

Issue: disaster37/rancher-backup#13

```text
2017-04-28 03:54:21,938 ERROR [:] [] [] [] [1020391880-8147] [i.g.i.g.r.handler.ExceptionHandler  ] Exception in API for request [https://rancher.wd-dev.com/v2-beta/stacks/1st3/]. Error id: [9890d141-8956-4a3e-aef9-ce3f5a3895be]. java.lang.NullPointerException: null
	at io.cattle.platform.servicediscovery.api.service.impl.RancherCertificatesToComposeFormatter.getCertName(RancherCertificatesToComposeFormatter.java:41) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.api.service.impl.RancherCertificatesToComposeFormatter.format(RancherCertificatesToComposeFormatter.java:29) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.api.service.impl.ServiceDiscoveryApiServiceImpl.translateRancherToCompose(ServiceDiscoveryApiServiceImpl.java:679) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.api.service.impl.ServiceDiscoveryApiServiceImpl.createComposeData(ServiceDiscoveryApiServiceImpl.java:183) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.api.service.impl.ServiceDiscoveryApiServiceImpl.buildRancherComposeConfig(ServiceDiscoveryApiServiceImpl.java:131) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.api.service.impl.ServiceDiscoveryApiServiceImpl.buildComposeConfig(ServiceDiscoveryApiServiceImpl.java:112) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.api.action.StackExportConfigActionHandler.perform(StackExportConfigActionHandler.java:57) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.api.resource.AbstractObjectResourceManager.resourceActionInternal(AbstractObjectResourceManager.java:382) ~[cattle-framework-api-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.resource.impl.AbstractBaseResourceManager.resourceAction(AbstractBaseResourceManager.java:414) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.resource.AbstractResourceManagerFilter.resourceAction(AbstractResourceManagerFilter.java:58) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.api.filter.ServiceStackStorageDriverFilter.resourceAction(ServiceStackStorageDriverFilter.java:85) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.resource.impl.FilteredResourceManager.resourceAction(FilteredResourceManager.java:86) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.resource.AbstractResourceManagerFilter.resourceAction(AbstractResourceManagerFilter.java:58) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.api.filter.ServiceStackNetworkDriverFilter.resourceAction(ServiceStackNetworkDriverFilter.java:85) ~[cattle-iaas-service-discovery-api-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.resource.impl.FilteredResourceManager.resourceAction(FilteredResourceManager.java:86) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.resource.AbstractResourceManagerFilter.resourceAction(AbstractResourceManagerFilter.java:58) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.resource.impl.FilteredResourceManager.resourceAction(FilteredResourceManager.java:86) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.handler.ResourceManagerRequestHandler.generate(ResourceManagerRequestHandler.java:44) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.handler.AbstractResponseGenerator.handle(AbstractResponseGenerator.java:14) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.DefaultReadWriteApiDelegate.handle(DefaultReadWriteApiDelegate.java:27) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.DefaultReadWriteApiDelegate.write(DefaultReadWriteApiDelegate.java:22) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at sun.reflect.GeneratedMethodAccessor540.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_72]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_72]
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:333) ~[spring-aop-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190) ~[spring-aop-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157) ~[spring-aop-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99) ~[spring-tx-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:280) ~[spring-tx-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96) ~[spring-tx-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92) ~[spring-aop-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:213) ~[spring-aop-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at com.sun.proxy.$Proxy66.write(Unknown Source) ~[na:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.ReadWriteApiHandler.handle(ReadWriteApiHandler.java:19) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.servlet.ApiRequestFilterDelegate.doFilter(ApiRequestFilterDelegate.java:99) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.api.servlet.ApiRequestFilter$1.runInContext(ApiRequestFilter.java:96) [cattle-framework-api-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.api.servlet.ApiRequestFilter.doFilter(ApiRequestFilter.java:106) [cattle-framework-api-0.5.0-SNAPSHOT.jar:na]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlets.UserAgentFilter.doFilter(UserAgentFilter.java:83) [jetty-servlets-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlets.GzipFilter.doFilter(GzipFilter.java:300) [jetty-servlets-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577) [jetty-security-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.Server.handle(Server.java:499) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_72]
```

**Comment by deniseschannon** [comment]
@willseward we made a fix in v1.5.7 and it will be in v1.6.0 as well. Let me know if you see this again. It's hard for us to reproduce but we made the fix based on your NPE.

**Comment by willseward** [comment]
@deniseschannon Thanks for the fix.
